scripts/evaluate.py

```diff
@@ -373,6 +373,166 @@ def validator(inputs):
+                padding=[3,3,3,3],strides=[2,2],
+                out_dtype=dtype)
+        D = relay.op.add(conv, bias)
+        D = relay.op.nn.relu(D)
+
+        mod = relay.Function([A, B, bias], D)
+        #mod, params = relay.testing.init.create_workload(func)
+        np.random.seed(0)
+        initializer = relay.testing.init.Xavier()
+        filter_data = np.zeros(filter_shape).astype(dtype)
+        bias_data = np.zeros(bias_shape).astype(dtype)
+        initializer("weight", filter_data)
+        initializer("bias", bias_data)
+        params = {
+            "weight": tvm.nd.array(filter_data),
+            "bias" : tvm.nd.array(bias_data),
+        }
+
+        def validator(inputs):
+            vec_length = input_shape[-1]
+            # nchwc -> nchw
+            data = inputs[0].transpose((0, 1, 4, 2, 3)).reshape(inputs[0].shape[0], inputs[0].shape[1]*inputs[0].shape[-1], inputs[0].shape[2], inputs[0].shape[3])
+            data = data.astype("float32")
+            # kcrsk -> kcrs
+            w_np = params["weight"].asnumpy()
+            kernel = w_np.transpose((0, 4, 1, 2, 3)).reshape(w_np.shape[0] * w_np.shape[4], w_np.shape[1], w_np.shape[2], w_np.shape[3])
+            np_result = testing.conv2d_nchw_python(data, kernel, padding=[1,1,1,1], stride=[1,1])
+            # nkhw -> nkhwk
+            np_result = np_result.reshape(np_result.shape[0], np_result.shape[1]//vec_length, vec_length, np_result.shape[2], np_result.shape[3]).transpose(0, 1, 3, 4, 2)
+            return [np_result,]
+        return (mod, params, {"data": input_shape}, dtype, target, validator)
+        #return (mod, params, {"data": input_shape}, dtype, target)
+
+    def import_conv2d_resnet50_v2_nchw_3c(self, target="llvm", dtype="float32"):
+        input_shape = (1, 3, 224, 224)
+        filter_shape = (64, 3, 7, 7)
+        bias_shape = (1, 64, 1, 1)
+        A = relay.var("data", shape=input_shape, dtype=dtype)
+        B = relay.var("weight", shape=filter_shape, dtype=dtype)
+        bias = relay.var("bias", shape=bias_shape, dtype=dtype)
+
+        #C = relay.nn.relu(A)
+        conv = relay.nn.conv2d(A, B, data_layout="NCHW", kernel_layout="OIHW",
+                            padding=[3,3,3,3],strides=[2,2],
+                            out_dtype=dtype, channels=64, kernel_size=(7,7))
+        D = relay.op.add(conv, bias)
+        D = relay.op.nn.relu(D)
+
+        mod = relay.Function([A, B, bias], D)
+        # mod, params = relay.testing.init.create_workload(func)
+        np.random.seed(0)
+        initializer = relay.testing.init.Xavier()
+        filter_data = np.zeros(filter_shape).astype(dtype)
+        bias_data = np.zeros(bias_shape).astype(dtype)
+        initializer("weight", filter_data)
+        initializer("bias", bias_data)
+        params = {
+            "weight": tvm.nd.array(filter_data),
+            "bias" : tvm.nd.array(bias_data),
+        }
+
+        # def validator(inputs):
```

Comment:
If validator is a dead code, we should remove it. Otherwise, is there any good reason to comment out?

Reply:
Done. Removed this code. We don't use validator here because it was necessary to modify it. We didn't do it because we were focused on the new featured development. Also, accuracy for all new layers is tested here: `tvm/tests/python/relay/test_conv2d_nchw_pack_texture.py`